src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -152,8 +167,17 @@ create_topic_properties_update(alter_configs_resource& resource) {
                 continue;
             }
             if (cfg.name == topic_property_remote_read) {
-                // Legacy update for shadow indexing field
-                {
+                if (shadow_indexing_split_update) {
```

Comment:
nit: can we get rid of 'shadow indexing' in the name? maybe just 'tiered-storage'?
the SI comes from the very first implementation which was called 'shadow indexing' and since then it was renamed everywhere.  I don't mind if the name will stay though. It will be easier to connect it with the old flag.

Reply:
also the 'split' in the name focuses on the low level detail, it will probably be confusing for some people

Reply:
Given that the topic property is still named `shadow_indexing`, I think it would be best if this rename was applied there as well in a future PR.